emData/generate_MC.py

```diff
@@ -0,0 +1,186 @@
+# dictionaries with the MC names as keys.
+wiresFile = open(wiresFileName)
+CMMems = {}
+FMMems = {}
+for line in wiresFile:
+    # Only barrel-only seeds are supported right now.
+    # Works for MC L3 phi sector C
+    if not any(["MC_" + l + "PHIB" in line for l in ["L3", "L4", "L5", "L6"]]) and \
+       not any(["MC_" + l + "PHIC" in line for l in ["L3"]]): #MC_L3PHIC was used for testing
+        continue
+    line = line.rstrip()
+    tcName = re.sub(r".*MC_(......).*", r"MC_\1", line)
+    memName = line.split()[0]
+    if memName.startswith("CM_"):
+        if tcName not in CMMems:
+            CMMems[tcName] = []
+        CMMems[tcName].append(memName)
+    if memName.startswith("FM_"):
+        FM = re.sub(r"FM_(....)_......", r"\1", memName)
+        if tcName not in FMMems:
+            FMMems[tcName] = []
+        FMMems[tcName].append(FM)
+wiresFile.close()
+
+# Open and print out preambles for the parameters and top files.
+dirname = os.path.dirname(os.path.realpath('__file__'))
+parametersFile = open(os.path.join(dirname, "../TrackletAlgorithm/MatchCalculator_parameters.h"), "w")
+parametersFile.write(
+    "#ifndef TrackAlgorithm_MatchCalculator_parameters_h\n"
+    "#define TrackAlgorithm_MatchCalculator_parameters_h\n"
+    "\n"
+    "// This file contains numbers of memories and bit masks that are specific to\n"
+    "// each MatchCalculator and that come directly from the wiring.\n"
+    "//\n"
+    "// The validity of each of the barrel CM memories is determined by\n"
+    "// FMMask. The bits of this mask, from least significant to most\n"
+    "// significant, represent the memories in the order they are passed to\n"
+    "// MatchCalculator; e.g., the LSB corresponds to\n"
+    "// TF::L1L2. If a bit is set, the corresponding memory is\n"
+    "// valid, if it is not, the corresponding memory is not valid.\n"
+)
+topHeaderFile = open(os.path.join(dirname, "../TrackletAlgorithm/MatchCalculatorTop.h"), "w")
+topHeaderFile.write(
+    "#ifndef TrackAlgorithm_MatchCalculatorTop_h\n"
+    "#define TrackAlgorithm_MatchCalculatorTop_h\n"
+    "\n"
+    "#include \"MatchCalculator.h\"\n"
+    "\n"
+    "constexpr int maxMatchCopies(" + str(len(TF_index)) + ");\n"
+    "constexpr int maxFullMatchCopies(" + str(len(TF_index)) + ");\n"
+)
+topFile = open(os.path.join(dirname, "../TrackletAlgorithm/MatchCalculatorTop.cc"), "w")
+topFile.write(
+    "#include \"MatchCalculatorTop.h\"\n"
+    "\n"
+    "////////////////////////////////////////////////////////////////////////////////\n"
+    "// Top functions for various MatchCalculators (MC). For each iteration of\n"
+    "// the main processing loop, a MC retrieves an array of Candidate Matches,\n"
+    "// and computes Full Matches from each.\n"
+    "////////////////////////////////////////////////////////////////////////////////\n"
+)
+
+# Calculate parameters and print out parameters and top function for each MC.
+for tcName in sorted(CMMems.keys()):
```

Comment:
@bryates Please change `tcName` → `mcName`.

Reply:
Done. I'll push the change soon.